### ISSUE TITLE
[v2 FEATURE]: Alter dropout for foundation model at train or hpopt time

### ISSUE THREAD

**Issue by lewismervin1** [chemprop/chemprop#1302]
**Is your feature request related to a problem? Please describe.**
At the moment dropout must be the same for the message_passing of any derived models from foundation models.  It is also ignored by hopt during hyperparameter optimization, despite dropout as being a parameter influencing the FFN.

E.g. for train, the mp_block and aggregation are loaded from the foundation model and not constructed with CLI args, so args.dropout is never read for message passing, e.g. one example is here:

https://github.com/chemprop/chemprop/blob/main/chemprop/cli/train.py#L1246

and for hpopt it is discarded:

https://github.com/chemprop/chemprop/blob/main/chemprop/cli/hpopt.py#L310-L319

It would be nice to have greater downstream flexibility of dropout when applied to foundation models.

I understand altering dropout betwen pre-train and fine-tune time may degrade performance or add noise to well-regularized foundation models / large fine-tuned datasets. However, it may have use if you wanting to regularize to smaller or domain-shifted datasets.

If not for the `message_passing`, think it would certainly be helpful to add dropout to hpopt for the `predictor`, so that optimal regularization of the FFN for fine-tuning can be evaluated?

**Use-cases/examples of this new feature**
Users can alter the dropout between foundation pre-train time, and fine-tune time. And users can optimize the optimal FFN dropout for fine-tuning.

**Desired solution/workflow**
Allow dropout to pass through hpopt, and to influence the `message_passing` and/or `predictor` during evaluation

**Comment by JacksonBurns** [comment]
I'm going to close this issue, since the answer I will provide in issues #1305 will also answer this one.
